### PR TITLE
porting in bfly updates

### PR DESCRIPTION
**Pull request by davidcairns**

### CODE REVIEW

File: Sources/ComposableArchitecture/Store.swift [hunk 1]
```diff
@@ -105,14 +222,16 @@ public final class Store<State, Action> {
   ///     `LocalState`.
   ///   - fromLocalAction: A function that transforms `LocalAction` into `Action`.
   /// - Returns: A publisher of stores with its domain (state and action) transformed.
-  public func scope<LocalState, LocalAction>(
-    state toLocalState: @escaping (Observable<State>) -> Observable<LocalState>,
+  public func scope<O: ObservableType, LocalState, LocalAction>(
```
Comment (davidcairns) [inline Sources/ComposableArchitecture/Store.swift:225]: Note that in PF’s upstream code this and the other slightly-tweaked overrides of `scope` are now called `publisherScope`.

File: Sources/ComposableArchitecture/Reducer.swift [hunk 2]
```diff
@@ -339,4 +464,12 @@ public struct Reducer<State, Action, Environment> {
   ) -> Effect<Action> {
     self.reducer(&state, action, environment)
   }
+
+  func receive(on scheduler: SchedulerType) -> Self {
```
Comment (kurtguenther) [inline Sources/ComposableArchitecture/Reducer.swift]: Can this be done in an extension in a separate class to provide simpler diffs in the future?
Also some docs would be nice at least for what problem this is solving.

Comment (kurtguenther) [reply Sources/ComposableArchitecture/Reducer.swift]: I get the other changes which require private vars cannot b.e

**Review by kurtguenther — APPROVED** [review]

File: Sources/ComposableArchitecture/Reducer.swift (hunk 2 above)
Comment (davidcairns) [reply Sources/ComposableArchitecture/Reducer.swift]: Good note; Moved this implementation into `Store+Scheduling`.

**Pull request merged**